types/budget.d.ts

```diff
@@ -39,8 +41,11 @@ declare global {
       /** Supported timing metrics. */
       export type TimingMetric = 'first-contentful-paint' | 'first-cpu-idle' | 'interactive' | 'first-meaningful-paint' | 'estimated-input-latency';
 
-      /** Supported resource types. */
+      /** These are the possible values for NetworkRequest.resourceType. */
       export type ResourceType = 'stylesheet' | 'image' | 'media' | 'font' | 'script' | 'document' | 'other';
+
+      /** Supported values for the resourceType property of a ResourceBudget. These are the values that a user can user in their budget.json. Most budget code should probably use this and not ResourceType */
+      export type BudgetResourceType =  ResourceType | 'total' | 'third-party';
```

Comment:
my assumption was that there was some value in splitting them because for example `determineResourceType` never used 'total' or 'third-party' and all the code that came before that used this type 'total' and 'third-party' weren't available options

if that's not true, +1 to merging back and apologies for the noise :)

Reply:
I just switched it back. #determineResourceType does use it, but that's the only use case I can think of in the code, even once the other branches are added.